windrecorder/flag_mark_note.py

```diff
@@ -0,0 +1,221 @@
+        self.resizable(False, False)  # 移除窗口放大选项
+        self.iconbitmap("__assets__\\icon-tray.ico")
+        FONT_CONFIG = ("Microsoft YaHei", 13)
+        customtkinter.set_appearance_mode("system")
+
+        # 添加标记后的提示项
+        self.label_added = customtkinter.CTkLabel(
+            self, text="✔ " + _t("flag_text_mark_added"), fg_color="transparent", font=FONT_CONFIG
+        )
+        self.label_added.grid(row=0, column=0, padx=15, pady=5, sticky="w")
+        self.label_time = customtkinter.CTkLabel(
+            self,
+            text=datetime.datetime.strftime(datetime_input, "%Y-%m-%d %H:%M:%S"),
+            fg_color="transparent",
+            text_color="#878787",
+            font=FONT_CONFIG,
+        )
+        self.label_time.grid(row=0, column=3, padx=15, pady=0, sticky="e")
+
+        # 备注输入框
+        self.textbox = customtkinter.CTkTextbox(master=self, height=80, font=FONT_CONFIG)
+        self.textbox.grid(row=2, column=0, columnspan=4, padx=5, pady=5, sticky="ew")
+        self.textbox.insert("0.0", _t("flag_input_note"))
+
+        # 移除标记 按钮
+        self.button_remove = customtkinter.CTkButton(
+            self,
+            text="❌ " + _t("flag_btn_remove_mark"),
+            command=lambda: self.delete_record(datetime_input=datetime_input),
+            width=0,
+            fg_color=("#DFDDDD", "#3D3D3D"),
+            text_color=("#3A3A3A", "#E7E7E7"),
+            hover_color=("#d4cdc8", "#303030"),
+            font=FONT_CONFIG,
+        )
+        self.button_remove.grid(row=3, column=0, padx=5, pady=5, columnspan=1, sticky="ew")
+
+        # 添加标记 按钮
+        self.button_add_note = customtkinter.CTkButton(
+            self,
+            text="✔ " + _t("flag_btn_add_note"),
+            command=lambda: self.add_note(datetime_input=datetime_input),
+            fg_color=("#E5DBFB", "#8262c9"),
+            text_color=("#4B357E", "#ffffff"),
+            hover_color=("#cfbef6", "#6f53af"),
+            font=FONT_CONFIG,
+        )
+        self.button_add_note.grid(row=3, column=1, padx=5, pady=5, columnspan=3, sticky="ew")
+
+    def add_note(self, datetime_input):
+        """
+        添加标记按钮 的回调
+        """
+        user_input_note = self.textbox.get("1.0", "end-1c")
+        # print(f"{user_input_note=}")
+        # "1.0"：表示从第一行的第一个字符开始获取文本。
+        # "end-1c"：表示获取到文本的倒数第二个字符为止，这样可以避免获取到最后的换行符。
+        if user_input_note.startswith(_t("flag_input_note")):  # 移除输入框中的提示前缀
+            user_input_note = user_input_note.replace(_t("flag_input_note"), "")
+
+        add_note_to_csv_by_datetime(user_input_note, datetime_input)
+        self.destroy()
+
+    def delete_record(self, datetime_input):
+        """
+        删除标记按钮 的回调：删除对应时间的记录
+        """
+        df = file_utils.read_dataframe_from_path(config.flag_mark_note_filepath)
+        df = df.drop(df[df["datetime"] == str(datetime.datetime.strftime(datetime_input, "%Y-%m-%d %H:%M:%S"))].index)
```

Comment:
What happen if there's no record with `datetime_input`

Reply:
`datetime_input` is a must have required item managed by the program. The class method will hold `datetime_input` until user choose to close window or add/delete note, at this point the whole process is closed/and end. Among the expectations I can think of, problems may only occur when user manually delete csv file after the window pop up. Even so, if `df` not found a matching datetime, there should be no exception raised (will just ignored